### PR TITLE
[BACKLOG-12852] Lat Long member properties are now assigned to the us…

### PR DESCRIPTION
**Pull request by DFieldFL**
…er defined Geo Location and removed from the auto modeled location if it exists. If auto modeled lat/long values have been removed then search all the fields againg to see if they exist.

### CODE REVIEW

**Comment by DFieldFL** [comment]
@pentaho/ackbar

File: core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java [hunk 1]
```diff
@@ -338,4 +337,79 @@ public GeoRole getGeoRoleByName( String name ) {
...
+
+      LocationRole locationRole = getLocationRole();
+
+      boolean locationFieldDetected = false;
+      int latColIndex = 0;
+      int lonColIndex = 0;
+      int count = 0;
+      // must iterate over the physical columns to ensure we process the columns in the proper order, available fields
+      // are sorted in available table
+      for ( IPhysicalColumn col : table.getPhysicalTable().getPhysicalColumns() ) {
```
Comment (krivera-pentaho) [inline core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java]: Null check on the "physical table"

File: core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java [hunk 2]
```diff
@@ -338,4 +337,79 @@ public GeoRole getGeoRoleByName( String name ) {
...
+    locationLevel.getMemberAnnotations().put( ANNOTATION_DATA_ROLE, locationRole );
+    locationLevel.getMemberAnnotations().put( ANNOTATION_GEO_ROLE, locationRole );
+  }
+
+  private void setLocationMemberProperties( ModelerWorkspace workspace, AvailableTable table,
+                                            LevelMetaData locationLevel, int latColIndex, int lonColIndex ) {
+    // if it is a LocationField we need to make sure the lat & long columns get
+    // added as logical columns to the model.
+    AvailableField latField =
+      table.findFieldByPhysicalColumn( table.getPhysicalTable().getPhysicalColumns().get( latColIndex ) );
```
Comment (krivera-pentaho) [inline core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java]: Null check on physical table and physical columns. Same comment below calls.

File: core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java [hunk 3]
```diff
@@ -338,4 +337,79 @@ public GeoRole getGeoRoleByName( String name ) {
...
+    AvailableField lonField =
+      table.findFieldByPhysicalColumn( table.getPhysicalTable().getPhysicalColumns().get( lonColIndex ) );
+
+    ColumnBackedNode tmp = workspace.createColumnBackedNode( latField, ModelerPerspective.ANALYSIS );
+    tmp.getLogicalColumn().setName( new LocalizedString( workspace.getWorkspaceHelper().getLocale(), LATITUDE ) );
```
Comment (krivera-pentaho) [inline core/src/main/java/org/pentaho/agilebi/modeler/geo/GeoContext.java]: Same comment here

File: core/src/main/java/org/pentaho/agilebi/modeler/models/annotations/AnnotationType.java [hunk 4]
```diff
@@ -347,6 +351,21 @@ protected LevelMetaData locateLevel( final ModelerWorkspace workspace, final Str
     return null;
   }
 
+  protected LevelMetaData locateLocationLevel( final ModelerWorkspace workspace ) throws ModelerException {
+    workspace.getModel().getDimensions();
```
Comment (krivera-pentaho) [inline core/src/main/java/org/pentaho/agilebi/modeler/models/annotations/AnnotationType.java]: NPE possibility here as well and below

File: core/src/main/java/org/pentaho/agilebi/modeler/models/annotations/CreateAttribute.java [hunk 5]
```diff
@@ -463,7 +467,29 @@ private void fillLevelProperties( final ModelerWorkspace workspace, final Logica
         removeAutoGeo( workspace );
       }
       GeoRole geoRole = workspace.getGeoContext().getGeoRoleByName( getGeoType().name() );
-      levelMetaData.getMemberAnnotations().put( "Data.Role", geoRole );
+      levelMetaData.getMemberAnnotations().put( ANNOTATION_DATA_ROLE, geoRole );
```
Comment (krivera-pentaho) [inline core/src/main/java/org/pentaho/agilebi/modeler/models/annotations/CreateAttribute.java]: NPE here

File: core/src/test/java/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java [hunk 6]
```diff
@@ -258,4 +259,15 @@ public static Document getMondrianDoc( final String mondrianTestFilePath )
       .newDocumentBuilder()
       .parse( mondrianSchemaXmlFile );
   }
+
+  public static OlapAnnotation getOlapAnnotationByName( final String name, List<OlapAnnotation> annotations) {
+    OlapAnnotation foundAnnotation = null;
+    for( OlapAnnotation annotation : annotations ) {
+      if ( name.equals( annotation.getName() ) ) {
+        foundAnnotation = annotation;
```
Comment (krivera-pentaho) [inline core/src/test/java/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java]: break from for loop if annotation is found

Comment (DFieldFL) [inline core/src/test/java/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java]: Check latest commit

**Pull request merged**